Frameworks/CoreText/DWriteWrapper.mm

```diff
@@ -493,6 +507,42 @@ static void __GetGlyphRunDetails(_CTTypesetter* ts, CFRange range, CGRect frameS
                 line->_width += glyphRunDetails._dwriteGlyphRun[i].glyphAdvances[index];
             }
 
+            DWRITE_GLYPH_METRICS glyphMetrics[run->_dwriteGlyphRun.glyphCount];
```

Comment:
Is there any reason to compute these upfront?  If the caller is not using these values, it would be a waste of time.  Rather initialize these from CTLineGetTypographicBounds?